### PR TITLE
Loosen mixlib-shellout/ohai deps and upgrade chefstyle

### PR DESCRIPTION
**Pull request by tas50**
Loosen pins to allow for the latest deps.

Signed-off-by: Tim Smith <tsmith@chef.io>

### CODE REVIEW

File: .rubocop.yml [hunk 1]
```diff
@@ -0,0 +1,852 @@
...
+Bundler/OrderedGems:
+  Exclude:
+    - 'Gemfile'
+
+# Cop supports --auto-correct.
+# Configuration parameters: TreatCommentsAsGroupSeparators, Include.
+# Include: **/*.gemspec
+Gemspec/OrderedDependencies:
+  Exclude:
+    - 'omnibus.gemspec'
```
Comment (lamont-granquist) [inline .rubocop.yml]: we don't ever want to enable these ones...

Comment (tas50) [reply .rubocop.yml]: I fixed that whole file up. I think I ran it with rubocop by accident

**Comment by lamont-granquist** [comment]
some sort of license_scout meltdown only on ruby 2.4

**Comment by lamont-granquist** [comment]
also on 2.5 on appveyor though

**Comment by tas50** [comment]
the new ~> heredoc is not behaving as expected there on Ruby 2.4. Really odd

**Comment by tas50** [comment]
Merging based on Slack approval

**Pull request merged**